### PR TITLE
LPS-116357 FAKE GRADLE CACHE

### PR DESCRIPTION
**Pull request by lawrence-lee**
Your explanation from the last PR makes sense. I'll add more questions I have inline.

### CODE REVIEW

**Comment by lawrence-lee** [comment]
ci:test:project-templates

File: modules/sdk/project-templates/project-templates-rest-builder/src/main/resources/archetype-resources/__artifactId__-test/bnd.bnd [hunk 1]
```diff
@@ -1,4 +1,8 @@
 Bundle-Name: ${artifactId}-test
 Bundle-SymbolicName: ${package}.test
 Bundle-Version: ${version}
--includeresource: ${package}.client.jar=${artifactId}-client-*.jar;lib:=true
\ No newline at end of file
+#if (${buildType} != "gradle")
+-includeresource: ${package}.client.jar=../${artifactId}-client/target/${artifactId}-client-${version}.jar;lib:=true
```
Comment (lawrence-lee) [inline modules/sdk/project-templates/project-templates-rest-builder/src/main/resources/archetype-resources/__artifactId__-test/bnd.bnd:5]: I know we're adding the support for maven, but does the *-test.jar work in maven?

Comment (lawrence-lee) [reply modules/sdk/project-templates/project-templates-rest-builder/src/main/resources/archetype-resources/__artifactId__-test/bnd.bnd:5]: If it doesn't work even if deployed, maybe we should remove it using the postGenerate task.

**Comment by lawrence-lee** [comment]
Also, can you add test coverage for the -test and -client jars? 

If we only support gradle, then add smaller unit tests to check the velocity and archetype generator the correct outputs for the test and client jars. The issue we saw in BLADE shouldve been caught here. 

If we support maven also, then in addition to the unit tests, we should also do a bundle diff test like we do with the -api and -impl.

File: modules/sdk/project-templates/project-templates-rest-builder/src/main/resources/archetype-resources/__artifactId__-test/bnd.bnd (hunk 1 above)
Comment (javierdearcos) [reply modules/sdk/project-templates/project-templates-rest-builder/src/main/resources/archetype-resources/__artifactId__-test/bnd.bnd:5]: The truth is that we don't know if it is possible to run this kind of tests with Maven. The only reason to keep this folder in Maven is to ease a change in the build mechanism to gradle, but, as this seems uncommon, the best option may be to don't generate this module with Maven

**Comment by javierdearcos** [comment]
> Also, can you add test coverage for the -test and -client jars?
> 
> If we only support gradle, then add smaller unit tests to check the velocity and archetype generator the correct outputs for the test and client jars. The issue we saw in BLADE shouldve been caught here.
> 
> If we support maven also, then in addition to the unit tests, we should also do a bundle diff test like we do with the -api and -impl.

Sure! I will add new integration tests to check test and client generation

File: modules/sdk/project-templates/project-templates-rest-builder/src/main/resources/archetype-resources/__artifactId__-test/bnd.bnd (hunk 1 above)
Comment (javierdearcos) [reply modules/sdk/project-templates/project-templates-rest-builder/src/main/resources/archetype-resources/__artifactId__-test/bnd.bnd:5]: I was thinking about adding a flag to control if the user want to generate client and test modules or not. I'll try to add it and consider it in the generation tests

**Comment by javierdearcos** [comment]
Thank you very much for your help, fixes and feedback! 😄

**Pull request merged**